exporters/etw/include/opentelemetry/exporters/etw/etw_properties.h

```diff
@@ -0,0 +1,476 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "opentelemetry/version.h"
+
+#include "opentelemetry/common/key_value_iterable_view.h"
+
+#include <opentelemetry/nostd/span.h>
+#include <map>
+#include <string>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace core
+{};
+namespace trace
+{};
+namespace exporter
+{
+namespace ETW
+{
+
+/**
+ * @brief PropertyVariant provides:
+ * - a constructor to initialize from initializer lists
+ * - an owning wrapper around `common::AttributeValue`
+ */
+using PropertyVariant =
+    nostd::variant<bool,
+                   int32_t,
+                   int64_t,
+                   uint32_t,
+                   uint64_t,
+                   double,
+                   std::string,
+#ifdef HAVE_CSTRING_TYPE
+                   const char *,
+#endif
+#ifdef HAVE_SPAN_BYTE
+                   // TODO: 8-bit byte arrays / binary blobs are not part of OT spec yet!
+                   // Ref: https://github.com/open-telemetry/opentelemetry-specification/issues/780
+                   std::vector<uint8_t>,
+#endif
+                   std::vector<bool>,
+                   std::vector<int32_t>,
+                   std::vector<int64_t>,
+                   std::vector<uint32_t>,
+                   std::vector<uint64_t>,
+                   std::vector<double>,
+                   std::vector<std::string>>;
+
+/**
+ * @brief PropertyValue class that holds PropertyVariant and
+ * provides converter for non-owning common::AttributeValue
+ */
+class PropertyValue : public PropertyVariant
+{
+
+  /**
+   * @brief Convert span<T> to vector<T>
+   * @tparam T
+   * @param source
+   * @return
+   */
+  template <typename T>
+  static std::vector<T> to_vector(const nostd::span<const T, std::size_t(-1)> &source)
+  {
+    return std::vector<T>(source.begin(), source.end());
+  };
+
+  /**
+   * @brief Convert span<string_view> to vector<string>
+   * @param source
+   * @return
+   */
+  std::vector<std::string> static to_vector(const nostd::span<const nostd::string_view> &source)
+  {
+    std::vector<std::string> result;
+    result.reserve(source.size());
+    for (const auto &item : source)
+    {
+      result.push_back(std::string(item.data()));
```

Comment:
I sort of agree (partially). Let me digest this.. Somewhere I agree with the `.size()`, but not because of NULs in between. We want all incoming string properties be UTF-8, thus no nulls in the middle of a string.

For the byte arrays - it'd be best to have this spec issue resolved, to allow for byte arrays. I added feature flag for that, so that even before spec issue is approved - it's possible to compile the SDK with byte arrays support:
https://github.com/open-telemetry/opentelemetry-specification/issues/780

Also if we add bytes support now ahead of spec, we can:
- export byte arrays as int for OTLP. This does not break the spec.
- export byte arrays as byte arrays for ETW. It supports `win:Binary` byte array type [described here](https://docs.microsoft.com/en-us/windows/win32/wes/eventmanifestschema-inputtype-complextype)

Reply:
> We want all incoming string properties be UTF-8, thus no nulls in the middle of a string.

Agreed. Then I'll narrow my concern to strings that are not null terminated.